plugins/parsers/prometheusremotewrite/parser.go

```diff
@@ -0,0 +1,101 @@
+package prometheusremotewrite
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/gogo/protobuf/proto"
+	"github.com/prometheus/common/model"
+)
+
+type Parser struct {
+	DefaultTags map[string]string
+}
+
+func (p *Parser) Parse(buf []byte) ([]telegraf.Metric, error) {
+	var metrics []telegraf.Metric
+	var err error
+	var req prompb.WriteRequest
+
+
+	if err := proto.Unmarshal(buf, &req); err != nil {
+		return nil, fmt.Errorf("unable to unmarshal request body: %s", err)
+	}
+
+
+	now := time.Now()
+
+	for _, ts := range req.Timeseries {
+		tags := map[string]string{}
+		for key, value := range p.DefaultTags {
+			tags[key] = value
+		}
+
+		for _, l := range ts.Labels {
+			tags[l.Name] = l.Value
+		}
+
+		metricName := tags[model.MetricNameLabel]
```

Comment:
Can we be absolutely sure that tag exists? I will sleep better if we check that `model.MetricNameLabel` exists in `tags` by
```suggestion
		if _, ok := tags[model.MetricNameLabel]; !ok {
				return fmt.Errorf("metric name %q not found in tag-set", model.MetricNameLabel)
		}
		metricName := tags[model.MetricNameLabel]
```